Experimental GBFS 3.0 support

flaktack opened the pull request:
### Summary

Adds support for GBFS 3.0: the existing code for converting GBFS 2.3 to the internal data model is copied, and where possible the common parts are extracted into shared classes.

### Issue

#6708 

### Unit tests

The existing tests are extended for GBFS 3.0

### Documentation

TOD

### Changelog

TODO

### Bumping the serialization version id

TODO

flaktack committed:
GBFS 3.0 support

leonardehrenfried commented on application/src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/gbfs/v3_0/GbfsStationInformationMapperV30.java:
```diff
@@ -0,0 +1,111 @@
+          GBFSShortName::getText
+        )
+      )
+      // TODO: Convert geometry
+      // .withStationArea(station.getStationArea())
```
```suggestion
```

leonardehrenfried reviewed (dismissed):
I have thought about this. While I'm not a fan of the massive duplication, I believe the other options (more wrapping and/or generics) are probably worse and you're not saving any code. For this reason I approve.

leonardehrenfried commented:
One thing we should debate in the dev meeting: do we want the version suffixes in the class names even though we also have them in the package? I'm in favour of not using the suffix but I don't think there is a consensus on that.

Lastly, can you update the docs? https://docs.opentripplanner.org/en/latest/GBFS-Config/

flaktack committed:
GBFS: remove version suffixes from classes

Only a single class (GbfsFeedLoaderAndMapper) uses both versions of
GBFS, so using full import where required is acceptable.

flaktack committed:
Update GFS documentation with version 3.0

flaktack committed:
Merge remote-tracking branch 'upstream/dev-2.x' into feature/gbfs-3-0

flaktack commented:
I've updated the documentation and removed the version suffix from classes. Only one class needed to use both versions of the GBFS classes, where using the fully qualified name for the duplicated classes seemed reasonable.

leonardehrenfried committed:
Update application/src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/gbfs/v3_0/GbfsStationInformationMapper.java

flaktack committed:
GTBFS 3.0: remove special date parsing

flaktack committed:
GTBFS: use major version for packages

flaktack committed:
Merge remote-tracking branch 'upstream/dev-2.x' into feature/gbfs-3-0

leonardehrenfried commented:
I'm not seeing any change yet.

testower reviewed (approved):
I tested this with the service discovery sandbox and at least it works as advertised there!
